src/soundio/soundmanager.cpp

```diff
@@ -404,20 +400,20 @@ SoundDeviceError SoundManager::setupDevices() {
             }
         }
         QList<AudioOutput> outputs =
-                m_config.getOutputs().values(device->getInternalName());
+                m_config.getOutputs().values(pDevice->getInternalName());
 
         // Statically connect the Network Device to the Sidechain
-        if (device->getInternalName() == kNetworkDeviceInternalName) {
+        if (pDevice->getInternalName() == kNetworkDeviceInternalName) {
             AudioOutput out(AudioPath::RECORD_BROADCAST, 0, 2, 0);
             outputs.append(out);
             if (m_config.getForceNetworkClock()) {
-                pNewMasterClockRef = device;
+                pNewMasterClockRef = pDevice;
             }
         }
 
-        foreach (AudioOutput out, outputs) {
+        for (const auto& out: outputs) {
```

Comment:
deep copy

Just one of many examples, I won't mention any further. That's why I don't like implicit sharing. It seems to be convenient at first sight, but is a devil in disguise in the long run.